egui_plot/src/items/mod.rs

```diff
@@ -557,21 +653,113 @@ impl PlotItem for Line<'_> {
         let mut final_stroke: PathStroke = (*stroke).into();
         // if we have a gradient color, we need to wrap the stroke callback to transpose the position to a value
         // the caller can reason about
-        if let Some(gradient_callback) = self.gradient_color.clone() {
-            let local_transform = *transform;
-            let wrapped_callback = move |_rec: Rect, pos: Pos2| -> Color32 {
-                let point = local_transform.value_from_position(pos);
-                gradient_callback(point)
+        if let Some(callback) = gradient_color.clone() {
+            let local_tf = *transform;
+            let wrapped =
+                move |_r: Rect, p: Pos2| -> Color32 { callback(local_tf.value_from_position(p)) };
+            final_stroke = PathStroke::new_uv(stroke.width, wrapped);
+        }
+
+        if let Some(blocks) = blocks_xy {
+            if blocks.xs.is_empty() {
+                return;
+            }
+
+            let mut draw_one_block = |xs: &[f64], ys: &[f64]| {
```

Comment:
Can you factor this out and unify as much as possible with the previous single-segment case?